python/Ganga/Core/GangaRepository/SessionLock.py

```diff
@@ -369,48 +361,49 @@ def startup(self):
 
         # setup global lock
         self.global_lock_setup()
-        self.global_lock_acquire()
-        try:
-            # setup counter file if it does not exist, read it if it does
-            if not os.path.exists(self.cntfn):
-                try:
-                    fd = self.delay_init_open(self.cntfn)
-                    try:
-                        os.write(fd, "0")
-                    finally:
-                        os.close(fd)
-                    #registerGlobalSessionFile( self.cntfn )
-                except OSError as x:
-                    if x.errno != errno.EEXIST:
-                        raise RepositoryError(
-                            self.repo, "OSError on count file create: %s" % x)
-            try:
-                self.count = max(self.count, self.cnt_read())
-            except ValueError, err:
-                logger.debug("Startup ValueError Exception: %s" % err)
-                logger.error("Corrupt count file '%s'! Trying to recover..." % (self.cntfn))
-            except OSError as err:
-                logger.debug("Startup OSError Exception: %s" % err)
-                raise RepositoryError(self.repo, "OSError on count file '%s' access!" % (self.cntfn))
-            self.cnt_write()
-            # Setup session file
+
+        self.finish_startup()
+
+    @global_disk_lock
+    def finish_startup(self):
+        # setup counter file if it does not exist, read it if it does
+        if not os.path.exists(self.cntfn):
+            fd = None
             try:
-                fd = self.delay_init_open(self.fn)
-                try:
-                    os.write(fd, pickle.dumps(set()))
-                finally:
+                fd = os.open(self.cntfn, os.O_EXCL | os.O_CREAT | os.O_WRONLY)
```

Comment:
Because opening the file with a `with open` presents issues. I tried this it didn't work I didn't have the time to invest fixing it. This is what was done before. is it a requirement or an issue? Frankly I'm more interested in cleaning up other code here in this PR